### DISCUSSION TITLE
How to show a modal in a form from a ToggleColumn based on state?

### DISCUSSION THREAD

**Discussion by rappasoft** [filamentphp/filament#8581] [General]
I have a ToggleColumn, and when the toggle is turned off, I want to show a modal with a text box, and then capture that value in `updateStateUsing`, is that possible?

```php
ToggleColumn::make('my_toggle')
  ->label('My Toggle')
  ->updateStateUsing(function (bool $state, MyModel $record) {
      $record->update(['status' => $state]);

      if ($state === false) {
            // Show modal, get value of text box in modal
            
            $record->update(['reason' => $textBoxValue]);
      }
  })
```

I can't find anything about it in the docs, or i'm looking in the wrong place.

**Comment by leandrocfe** [comment]
What do you think about using a [custom action](https://filamentphp.com/docs/3.x/actions/modals#modal-forms) instead of using toggle column?
